src/Val.cc

```diff
@@ -1957,6 +2039,14 @@ Val* TableVal::Delete(const HashKey* k)
 	delete v;
 
 	Modified();
+
+	if ( change_func )
+		{
+		auto index = table_hash->RecoverVals(k);
+		CallChangeFunc(index, va->Ref(), element_removed);
```

Comment:
Seems `va` is allowed to be null here and ref'ing it would segfault, so probably mean to not take this branch at all in the "no such element found" case.